src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -836,7 +844,7 @@ class module_sram : public module_impl
   create_instr_buf(const module_impl* parent)
   {
     XRT_PRINTF("-> module_sram::create_instr_buf()\n");
-    auto data = parent->get_instr();
+    const instr_buf& data = parent->get_instr();
```

Comment:
Plus and minus here.  Is it clearer with the type in your face, maybe?
```suggestion
    const auto& data = parent->get_instr();
```